engine/runtime/src/main/java/org/enso/interpreter/node/callable/FunctionCallInstrumentationNode.java

```diff
@@ -57,23 +62,46 @@ public boolean isInstrumentable() {
   /** A simple value class for function call information. */
   @ExportLibrary(InteropLibrary.class)
   public static final class FunctionCall implements TruffleObject {
+    private final ModuleScope ctx;
     private final Function function;
-    private final Object state;
+    private final State state;
     private final @CompilerDirectives.CompilationFinal(dimensions = 1) Object[] arguments;
 
     /**
      * Creates an instance of this class.
      *
+     * @param ctx the scope where the instrumentation happens
      * @param function the function being called.
      * @param state the monadic state to pass to the function.
      * @param arguments the arguments passed to the function.
      */
-    public FunctionCall(Function function, Object state, Object[] arguments) {
+    public FunctionCall(ModuleScope ctx, Function function, State state, Object[] arguments) {
+      this.ctx = ctx;
       this.function = function;
       this.state = state;
       this.arguments = arguments;
     }
 
+    /**
+     * Creates an instance of this class.
+     *
+     * @param ctx the node where the instrumentation happens
+     * @param function the function being called.
+     * @param state the monadic state to pass to the function.
+     * @param arguments the arguments passed to the function.
+     */
+    public FunctionCall(Node ctx, Function function, State state, Object[] arguments) {
+      this(findModuleScope(ctx), function, state, arguments);
+    }
+
+    private static ModuleScope findModuleScope(Node ctx) {
+      return switch (ctx.getRootNode()) {
+        case EnsoRootNode root -> root.getModuleScope();
+        case null -> null;
```

Comment:
`getRootNode()` may return `null`. It shouldn't, but it may. `switch` statement would yield `NullPointerException` when that would happen without this `case null` line.

Reply:
gotta love java